snapshot/lib/snapshot/simulator_launchers/simulator_launcher_base.rb

```diff
@@ -116,7 +116,8 @@ def override_status_bar(device_type, arguments = nil)
       device_udid = TestCommandGenerator.device_udid(device_type)
 
       UI.message("Launch Simulator #{device_type}")
-      Helper.backticks("xcrun instruments -w #{device_udid} &> /dev/null")
+      # Boot the simulator and wait for it to finish booting
+      Helper.backticks("xcrun simctl bootstatus #{device_udid} -b &> /dev/null")
```

Comment:
Do you happen to have any good documentation or explanation on what the difference between these two commands are? I tried to look at the help/man pages and couldn't really find too much h🤷‍♂️

Reply:
With Xcode 12.5.1 as the current Xcode `xcrun instruments` outputs:

```
`instruments` is now deprecated in favor of 'xcrun xctrace' (see `man xctrace` for more information on its replacement)
instruments, version 12.5.1 (64544.132)
usage: instruments [-t template] [-D document] [-l timeLimit] [-w device] [-P package] [[-p pid] | [application [-e variable value] [argument ...]]]
```

`man instruments` describes `-w` as:

```
     -w [hardware device, or simulator name]
              The name or identifier of the hardware device or simulator to target. Should be specified first
```

(you probably found this documentation but I'm adding it for completeness)

The behaviour has either changed or this never worked because using instruments to boot the simulator only boots the simulator (it doesn't wait for it to be ready) and exits with an error:

```
instruments[2650]: Waiting for device to boot...
Instruments Usage Error: No template (-t) specified
instruments, version 12.5.1 (64544.132)
usage: instruments [-t template] [-D document] [-l timeLimit] [-w device] [-P package] [[-p pid] | [application [-e variable value] [argument ...]]]
```

---

`bootstatus` is not listed in `xcrun simctl help` but does have help text available via `xcrun simctl help bootstatus`:

```
Checks device boot status.
Usage: simctl bootstatus <device> [-bcd]
	-b       Boot the device if it isn't already booted.
	-d       Print data migration information, if it's available.
	-c       Continuously monitor boot status through multiple boot/shutdown cycles.

Monitors the specified device and prints boot status information until the device finishes booting. You can safely call this before you attempt to start booting the device.
```

---

I found this quite confusing myself.

Reply:
Oh my goodness! The `xcrun simctl help bootstatus` is the command I need to find but couldn't 😓 Thank you so much for this! It does appear that this never really worked then 😬

Reply:
It's a bit hidden! 😔 I would love to know if there are any other (hidden) commands available but this is the only one I'm aware of.

Reply:
Thanks for this tipp! This saved me from HOURS of debugging in my screenshotting tool!